src/main/java/uk/gov/hmcts/reform/sendletter/config/jpa/CustomRepositoryFactoryBean.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.hmcts.reform.sendletter.config.jpa;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.support.JpaRepositoryFactoryBean;
+import org.springframework.data.repository.core.support.RepositoryFactorySupport;
+import uk.gov.hmcts.reform.sendletter.logging.AppInsights;
+
+import java.io.Serializable;
+import javax.persistence.EntityManager;
+
+class CustomRepositoryFactoryBean<R extends JpaRepository<T, I>, T, I extends Serializable>
```

Comment:
I would recommend trying: https://github.com/hmcts/ccd-user-profile-api/blob/master/lib/applicationinsights-agent-2.3.1-SNAPSHOT.jar

This should do auto instrumentation of postgres, it was built by the platform engineering team for CCD, it includes one commit on top of 2.3.1
https://github.com/Microsoft/ApplicationInsights-Java/commit/00f384fa4781f39b52ed300b4fd8c2fa23968e1e